### PR TITLE
[HOPS-1483] Track changes on the metadaEnabled directory

### PR DESCRIPTION
**Pull request by maismail**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**
https://hopshadoop.atlassian.net/browse/HOPS-1483

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by berthoug** [comment]
Jenkins test this please.

**Comment by berthoug** [comment]
Jenkins test this please.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java [hunk 1]
```diff
@@ -920,6 +920,14 @@ boolean isPathMetaEnabled() throws TransactionContextException, StorageException
 
   INodeDirectory getMetaEnabledParent()
       throws TransactionContextException, StorageException {
+    
+    if(isDirectory()){
+      INodeDirectory self = (INodeDirectory) this;
```
Comment (berthoug) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java]: Wouldn't it be cleaner to overwrite the getMetaEnabledParent function in InodeDirectory?

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java [hunk 2]
```diff
@@ -1482,17 +1485,24 @@ private void setMetaEnabledInt(final String src,
     } else {
       INodeDirectory dirNode = (INodeDirectory) targetNode;
       dirNode.setMetaEnabled(metaEnabled);
+      if(metaEnabled){
+        dirNode.incrementLogicalTime();
+        INodeMetadataLogEntry dslogEntry =
+            new INodeMetadataLogEntry(dirNode.getId(),
+                dirNode.getId(), dirNode.getPartitionId(),
+        dirNode.getParentId(), dirNode.getLocalName(), dirNode.getLogicalTime(),
```
Comment (berthoug) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java]: can you reformat this

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java (hunk 1 above)
Comment (maismail) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java]: Yes, will fix that.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java (hunk 2 above)
Comment (maismail) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java]: ok

**Pull request merged**